plugins/opener/src/reveal_item_in_dir.rs

```diff
@@ -54,56 +96,79 @@ mod imp {
         },
     };
 
-    pub fn reveal_item_in_dir(path: &Path) -> crate::Result<()> {
-        let file = dunce::simplified(path);
+    pub fn reveal_items_in_dir(paths: &[PathBuf]) -> crate::Result<()> {
+        if paths.is_empty() {
+            return Ok(());
+        }
+
+        let first_path = dunce::simplified(&paths[0]);
+        let parent_dir = first_path
+            .parent()
+            .ok_or_else(|| crate::Error::NoParent(first_path.to_path_buf()))?;
+
+        // On Windows, SHOpenFolderAndSelectItems requires all items to be in the same directory.
+        // We filter the paths to ensure they all share the same parent as the first path.
+        let files_in_same_dir: Vec<_> = paths
+            .iter()
+            .map(|p| dunce::simplified(p))
+            .filter(|p| p.parent() == Some(parent_dir))
+            .collect();
+
+        if files_in_same_dir.is_empty() {
+            // This case can happen if the original list had paths from different directories.
+            // We can't open multiple directories, so we do nothing.
+            return Ok(());
+        }
 
         let _ = unsafe { CoInitialize(None) };
 
-        let dir = file
-            .parent()
-            .ok_or_else(|| crate::Error::NoParent(file.to_path_buf()))?;
+        let dir_hstring = HSTRING::from(parent_dir);
+        let dir_item = unsafe { ILCreateFromPathW(&dir_hstring) };
 
-        let dir = HSTRING::from(dir);
-        let dir_item = unsafe { ILCreateFromPathW(&dir) };
+        // Ensure dir_item is freed even if subsequent operations fail.
+        let mut created_file_items = Vec::new();
 
-        let file_h = HSTRING::from(file);
-        let file_item = unsafe { ILCreateFromPathW(&file_h) };
+        for path in &files_in_same_dir {
+            let file_hstring = HSTRING::from(path.as_os_str());
+            let file_item = unsafe { ILCreateFromPathW(&file_hstring) };
+            if !file_item.is_null() {
+                created_file_items.push(file_item);
```

Comment:
I think we could implement something like https://github.com/microsoft/windows-rs/blob/05879ec957df7ad92d0653fb84a83f75a0a9b4e9/crates/libs/core/src/handles.rs#L18 to make it lesser error prone to free

Reply:
it looks like it's fixed by your PR as well, right?